src/dogroups.c

```diff
@@ -153,34 +153,36 @@ SEXP dogroups(SEXP dt, SEXP dtcols, SEXP groups, SEXP grpcols, SEXP jiscols, SEX
           SET_VECTOR_ELT(VECTOR_ELT(SDall,j),0,R_NilValue);
           break;
         default:
-          error("Logical error. Type of column should have been checked by now");
+          error("Internal error. Type of column should have been checked by now"); // #nocov
         }
       }
       grpn = 1;  // it may not be 1 e.g. test 722. TODO: revisit.
       SETLENGTH(I, grpn);
       INTEGER(I)[0] = 0;
       for (int j=0; j<length(xSD); ++j) {
         switch (TYPEOF(VECTOR_ELT(xSD, j))) {
-        case LGLSXP :
-          LOGICAL(VECTOR_ELT(xSD,j))[0] = NA_LOGICAL;
-          break;
+        case LGLSXP : // #nocov
+          LOGICAL(VECTOR_ELT(xSD,j))[0] = NA_LOGICAL; // #nocov
+          break; // #nocov
```

Comment:
This is a case like in test 1540.36 (`on` column from `x` used in `j` during `by=.EACHI`)

IIUC `bmerge` will already have coerced `LGLSXP` to `INTSXP` by the time `dogroups` is working on it in this line, so it's not actually possible in the current code to reach this line.

I left as `nocov` but it could also be an `Internal error` as something to monitor?